tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp

```diff
@@ -162,6 +185,53 @@ void integrate_test_explicit_time_dependence(const TimeStepper& stepper,
   CHECK(history.size() < 20);
 }
 
+void integrate_error_test(const TimeStepper& stepper,
+                          const size_t number_of_past_steps,
+                          const double integration_time, const double epsilon,
+                          const size_t num_steps,
+                          const double error_factor) noexcept {
+  auto analytic = [](const double t) noexcept { return sin(t); };
+  auto rhs = [](const double v, const double /*t*/) noexcept {
+               return sqrt(1. - square(v));
+             };
+
+  const Slab slab = integration_time > 0
+      ? Slab::with_duration_from_start(0., integration_time)
+      : Slab::with_duration_to_end(0., -integration_time);
+  const TimeDelta step_size = integration_time > 0
+      ? slab.duration() / num_steps
+      : -slab.duration() / num_steps;
+
+  Time time = integration_time > 0 ? slab.start() : slab.end();
+  double y = analytic(time.value());
+  TimeSteppers::History<double, double> history;
+
+  initialize_history(time, make_not_null(&history), analytic, rhs, step_size,
+                     number_of_past_steps);
+  std::optional<double> y_error;
+  for (uint64_t i = 0; i < num_steps; ++i) {
+    take_step_and_check_error(make_not_null(&time), make_not_null(&y),
+                              make_not_null(&y_error), make_not_null(&history),
+                              stepper, rhs, step_size);
+    // This check needs a looser tolerance for lower-order time steppers.
+    CHECK(y == approx(analytic(time.value())).epsilon(epsilon));
+
+    // check that the error measure is a reasonable estimate of the deviation
+    // from the analytic solution. This solution is smooth, so the error should
+    // be dominated by the stepper.
+    if (i > num_steps / 2) {
```

Comment:
Why are you only checking the error estimate in the second half of the integration?

Reply:
I chose this because I'm requiring the stepper error measure to neither underestimate nor (significantly) overestimate the error to the analytic solution. In these tests, some of the higher-order steppers have very small errors near the start of the evolution of the sinusoid, so the error measure is a pretty drastic overestimate for the first several steps.
Is that behavior surprising? Do you think there's a better way to do this test reliably?